python/ray/workflow/common.py

```diff
@@ -183,9 +148,7 @@ class WorkflowInputs:
     args: ObjectRef
     # TODO(suquark): maybe later we can replace it with WorkflowData.
     # The workflows in the arguments.
-    workflows: "List[Workflow]"
-    # The dynamic refs of workflows in the arguments.
-    workflow_refs: List[WorkflowRef]
+    workflows_or_refs: "List[Union[Workflow, WorkflowRef]]"
```

Comment:
Should we have a class maybe called WorkflowArgs ?

In side this class, we'll just have one method to get the object ref/data (depends on the type).

WorkflowRef is just a WorkflowArg?

My feeling is that, for the Inputs, it should be abstracted to something common instead of exposing too many details about where does this input come from.
Where does this input come from and how it's resolved, should be handled by the actual implementation of the input itself.

Some high-level ideas that come to my mind, there are several type of inputs:

- It comes from Workflow (not executed)
- It comes from Workflow (executed)
- It's an py object (we may need to put first)
- it's an object ref

Can we use ObjectRef as the protocol communicated between components? For example, all these types will become an object ref and we use object ref as inputs.